### PR TITLE
feat(aws): Adding launch configurations cleanup support

### PR DESCRIPTION
**Pull request by jeyrschabu**
- added launch configs support

### CODE REVIEW

**Comment by jeyrschabu** [comment]
Delete logic coming next but this can be merged to collect some metrics.

**Comment by jeyrschabu** [comment]
Clouddriver op support: https://github.com/spinnaker/clouddriver/pull/4324

**Comment by jeyrschabu** [comment]
Orca support: https://github.com/spinnaker/orca/pull/3442

File: swabbie-aws/src/main/kotlin/com/netflix/spinnaker/swabbie/aws/launchconfigurations/AmazonLaunchConfigurationHandler.kt [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+        description = "Deleting LaunchConfigurations: $launchConfigurationNames"
+      )
+    ).let { taskResponse ->
+      val completeEvent = TaskCompleteEventInfo(
+        action = Action.DELETE,
+        markedResources = markedResources,
+        workConfiguration = workConfiguration,
+        submittedTimeMillis = clock.instant().toEpochMilli()
+      )
+
```
Comment (aravindmd) [inline swabbie-aws/src/main/kotlin/com/netflix/spinnaker/swabbie/aws/launchconfigurations/AmazonLaunchConfigurationHandler.kt]: Might be worth adding a log message with the resource / orca task id for debugging ?

File: swabbie-aws/src/main/kotlin/com/netflix/spinnaker/swabbie/aws/launchconfigurations/AmazonLaunchConfigurationHandler.kt [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+    val launchConfigurationNames = markedResources
+      .map {
+        it.resourceId
+      }.toSet()
+
+    orcaService.orchestrate(
+      OrchestrationRequest(
+        application = application,
+        job = listOf(
+          OrcaJob(
```
Comment (aravindmd) [inline swabbie-aws/src/main/kotlin/com/netflix/spinnaker/swabbie/aws/launchconfigurations/AmazonLaunchConfigurationHandler.kt]: do we need to introduce a wait like how we do for images ? not sure the # of launch configs we will start deleting

File: swabbie-aws/src/main/kotlin/com/netflix/spinnaker/swabbie/aws/launchconfigurations/AmazonLaunchConfigurationHandler.kt [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+   * Server Groups
+   */
+  private fun checkReferences(launchConfigurations: List<AmazonLaunchConfiguration>?, params: Parameters) {
+    if (launchConfigurations.isNullOrEmpty()) {
+      return
+    }
+
+    log.debug("checking references for {} launch configs. Parameters: {}", launchConfigurations.size, params)
+    val elapsedTimeMillis = measureTimeMillis {
+      setServerGroupReferences(launchConfigurations, params)
```
Comment (aravindmd) [inline swabbie-aws/src/main/kotlin/com/netflix/spinnaker/swabbie/aws/launchconfigurations/AmazonLaunchConfigurationHandler.kt]: can this throw an exception ? do we need to wrap this with a try/catch ?

Comment (jeyrschabu) [reply swabbie-aws/src/main/kotlin/com/netflix/spinnaker/swabbie/aws/launchconfigurations/AmazonLaunchConfigurationHandler.kt]: Exceptions would bubble up and get handled in `AbstractResourceTypeHandler`

File: swabbie-aws/src/main/kotlin/com/netflix/spinnaker/swabbie/aws/launchconfigurations/AmazonLaunchConfigurationHandler.kt (hunk 2 above)
Comment (jeyrschabu) [reply swabbie-aws/src/main/kotlin/com/netflix/spinnaker/swabbie/aws/launchconfigurations/AmazonLaunchConfigurationHandler.kt]: Let's collect some metrics first.

File: swabbie-aws/src/main/kotlin/com/netflix/spinnaker/swabbie/aws/launchconfigurations/AmazonLaunchConfigurationHandler.kt (hunk 1 above)
Comment (jeyrschabu) [reply swabbie-aws/src/main/kotlin/com/netflix/spinnaker/swabbie/aws/launchconfigurations/AmazonLaunchConfigurationHandler.kt]: That's a good idea. I would like to stick that in a common place like so it's called for every handler when a task is submitted. I will do that in a seperate PR.

**Pull request merged**